cirq-core/cirq/qis/clifford_tableau.py

```diff
@@ -326,32 +326,38 @@ def __str__(self) -> str:
         return string
 
     def _str_full_(self) -> str:
-        string = ''
-
-        string += 'stable' + ' ' * max(self.n * 2 - 3, 1)
-        string += '| destable\n'
-        string += '-' * max(7, self.n * 2 + 3) + '+' + '-' * max(10, self.n * 2 + 4) + '\n'
-
-        for j in range(self.n):
-            for i in [j + self.n, j]:
-                string += '- ' if self.rs[i] else '+ '
-
-                for k in range(self.n):
-                    if self.xs[i, k] & (not self.zs[i, k]):
-                        string += f'X{k}'
-                    elif (not self.xs[i, k]) & self.zs[i, k]:
-                        string += f'Z{k}'
-                    elif self.xs[i, k] & self.zs[i, k]:
-                        string += f'Y{k}'
-                    else:
-                        string += '  '
-
-                if i == j + self.n:
-                    string += ' ' * max(0, 4 - self.n * 2) + ' | '
-
-            string += '\n'
+        left_col_width = max(7, self.n * 2 + 3)
+        right_col_width = max(10, self.n * 2 + 4)
+
+        def fill_row(left: str, right: str, mid='|', fill=' ') -> str:
+            """Builds a left-aligned fixed-width row with 2 columns."""
+            return f"{left:{fill}<{left_col_width}}{mid}{right:{fill}<{right_col_width}}"
+
+        def pauli_from_matrix(r: int, c: int) -> str:
+            return {
+                (True, False): f'X{c}',
```

Comment:
Please rewrite using match-case statement - https://docs.python.org/3/whatsnew/3.10.html#patterns-with-a-literal-and-variable.  Should be more readable and avoid a temporary dictionary for each Pauli symbol resolution.